treeherder/intermittents_commenter/commenter.py

```diff
@@ -117,38 +119,41 @@ def generate_bug_changes(self, startday, endday, alt_startday, alt_endday):
                             }
                            }
 
-            if bug_info is not None and whiteboard != bug_info['whiteboard']:
-                bug_changes['changes']['whiteboard'] = whiteboard
+            if change_whiteboard:
+                bug_changes['changes']['whiteboard'] = change_whiteboard
+
             if change_priority:
                 bug_changes['changes']['priority'] = change_priority
 
             all_bug_changes.append(bug_changes)
 
         return all_bug_changes
 
-    def check_needswork_owner(self, change_priority, bug_info, whiteboard):
+    def check_needswork_owner(self, change_priority, bug_info):
+        change_priority = None
+        change_whiteboard = None
+
         if (([bug_info['product'], bug_info['component']] in COMPONENTS) and
-            not self.check_whiteboard_status(whiteboard)):
+            not self.check_whiteboard_status(bug_info['whiteboard'])):
 
             if bug_info['priority'] not in ['--', 'P1', 'P2', 'P3']:
                 change_priority = '--'
 
-            stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+            stockwell_text = re.search(r'\[stockwell (.+?)\]', bug_info['whiteboard'])
             # update whiteboard text unless it already contains WHITEBOARD_NEEDSWORK_OWNER
             if stockwell_text is None or stockwell_text.group() != WHITEBOARD_NEEDSWORK_OWNER:
-                whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_NEEDSWORK_OWNER)
+                change_whiteboard = self.update_whiteboard(bug_info['whiteboard'], WHITEBOARD_NEEDSWORK_OWNER)
 
-        return change_priority, whiteboard
+        return change_priority, change_whiteboard
 
     def check_needswork(self, whiteboard):
-        if whiteboard is not None:
-            stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
-            # update all [stockwell needswork] bugs (including all 'needswork' possibilities,
-            # ie 'needswork:owner') and update whiteboard to [stockwell unknown]
-            if stockwell_text is not None and stockwell_text.group(1).split(':')[0] == 'needswork':
-                whiteboard = self.update_whiteboard(whiteboard, '[stockwell unknown]')
+        stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+        # update all [stockwell needswork] bugs (including all 'needswork' possibilities,
+        # ie 'needswork:owner') and update whiteboard to [stockwell unknown]
+        if stockwell_text is not None and stockwell_text.group(1).split(':')[0] == 'needswork':
+            return self.update_whiteboard(whiteboard, '[stockwell unknown]')
 
-        return whiteboard
+        return None
 
     def assign_priority(self, priority, counts):
```

Comment:
You could eliminate the priority parameter here -- now that assign_priority is only ever called with priority == 0.

Reply:
Good point!